src/encoded/types/file.py

```diff
@@ -823,6 +823,8 @@ def get_open_data_url_or_presigned_url_location(self, external, request, filenam
             if hasattr(self.model, 'source'):
                 es_model_props = self.model.source['embedded']
                 open_data_url = es_model_props.get('open_data_url', None)
+                if filename not in open_data_url:  # we requested an extra_file, so recompute with correct filename
```

Comment:
I guess there is guaranteed to be a filename?  If not an exception could be thrown.